source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -2452,7 +2460,255 @@ export class MediaSource extends SerializableObject {
+            this._mediaElement = document.createElement("video");
+        } else {
+            this._mediaElement = document.createElement("audio");
+        }
+
+        this._mediaElement.setAttribute(
+            "aria-label",
+            this.owner.altText ? this.owner.altText : Strings.defaults.mediaPlayerAriaLabel()
+        );
+        this._mediaElement.setAttribute("webkit-playsinline", "");
+        this._mediaElement.setAttribute("playsinline", "");
+        this._mediaElement.autoplay = true;
+        this._mediaElement.controls = true;
+
+        if (Utils.isMobileOS()) {
+            this._mediaElement.muted = true;
+        }
+
+        this._mediaElement.preload = "none";
+        this._mediaElement.style.width = "100%";
+
+        for (const source of this.owner.sources) {
+            const renderedSource = source.render();
+
+            Utils.appendChild(this._mediaElement, renderedSource);
+        }
+
+        return this._mediaElement;
+    }
+
+    play() {
+        if (this._mediaElement) {
+            this._mediaElement.play();
+        }
+    }
+
+    get selectedMediaType(): string | undefined {
+        return this._selectedMediaType;
+    }
+}
+
+export abstract class CustomMediaPlayer extends MediaPlayer {
+    constructor(matches: RegExpExecArray) {
+        super();
+    }
+}
+
+export abstract class IFrameMediaMediaPlayer extends CustomMediaPlayer {
+    private _videoId?: string;
+
+    constructor(matches: RegExpExecArray, readonly iFrameTitle?: string) {
+        super(matches);
+
+        if (matches.length >= 2) {
+            this._videoId = matches[1];
+        }
+    }
+
+    abstract getEmbedVideoUrl(): string;
+
+    canPlay(): boolean {
+        return this._videoId !== undefined;
+    }
+
+    render(): HTMLElement {
+        let container = document.createElement("div");
+        container.style.position = "relative";
+        container.style.width = "100%";
+        container.style.height = "0";
+        container.style.paddingBottom = "56.25%";
+
+        let iFrame = document.createElement("iframe");
+        iFrame.style.position = "absolute";
+        iFrame.style.top = "0";
+        iFrame.style.left = "0";
+        iFrame.style.width = "100%";
+        iFrame.style.height = "100%";
+        iFrame.src = this.getEmbedVideoUrl();
+        iFrame.frameBorder = "0";
+
+        if (this.iFrameTitle) {
+            iFrame.title = this.iFrameTitle;
+        }
+
+        iFrame.allow = "accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture";
+        iFrame.allowFullscreen = true;
+        
+        container.appendChild(iFrame);
+
+        return container;
+    }
+
+    get videoId(): string | undefined {
+        return this._videoId;
+    }
+}
+
+export class VimeoPlayer extends IFrameMediaMediaPlayer {
+    async fetchVideoDetails(): Promise<void> {
+        const oEmbedUrl = `https://vimeo.com/api/oembed.json?url=${this.getEmbedVideoUrl()}`;
```

Comment:
Do you need to make the url in configuration? Likely 3rd party can't change it without code changing in our codebase. How about put it in customMediaPlayers

Reply:
I mean can we make `https://vimeo.com/api/oembed.json` in global definition or somewhere, then if vimeo changed their API endpoint, we don't need to make code change.

Reply:
I suppose we could always do something like that in the future, but I don't think Vimeo is likely to change their API as it would break a lot of client applications.